### PR TITLE
File structure refactoring, fixed AppUser streaming via Provider

### PR DESCRIPTION
**Pull request by tbeidlershenk**
### Changes
1. Changed folder structure to better categorize files and renamed some files / class names to match each other
2. Fixed the ```AppUser``` stream in ```MultiProvider``` in main.dart in order to stream the user's data from Firestore based on the auth user
3. Boy scout rule'd some files ~ most notably I made ```FirestoreService``` and ```AuthService``` to have only static methods (we were instancing these classes when we needed to use their methods, it didn't really make sense)
4. Updated Wrapper to map to LoadingAnimation if the AppUser Provider hasn't received data yet

### Notes
I think the way I did change # 4 is kind of bad let me know if you don't get what I did or have a better idea of how to do it. Also you can verify that the AppUser stream is working by looking in ProfilePage, it loads the first name of the user which is one of the fields that is only stored in Firestore, not in Firebase Auth

### CODE REVIEW

File: lib/data/app_user.dart [hunk 1]
```diff
@@ -1,29 +1,36 @@
...
           email: data['email'],
           firstName: data['firstName'],
           lastName: data['lastName'],
           location: data['location']);
     } catch (e) {
-      return MyUser(
+      return AppUser(
           uid: 'UserID',
           email: 'someone@example.com',
           firstName: "First",
```
Comment (jeffzheng13) [inline lib/data/app_user.dart:36]: Should we just return default user here instead?

File: lib/services/auth_service.dart [hunk 2]
```diff
@@ -1,67 +1,61 @@
...
     } catch (e) {
       print(e.toString());
-      return null;
+      return false;
     }
   }
 
   //sign in with email and password
-  Future signInWithEmailAndPassword(String _email, String _password) async {
+  static Future<void> signInWithEmailAndPassword(
```
Comment (jeffzheng13) [inline lib/services/auth_service.dart:45]:
```suggestion
  static Future signInWithEmailAndPassword(
```

File: lib/services/auth_service.dart [hunk 3]
```diff
@@ -1,67 +1,61 @@
...
     UserCredential result = await _auth.signInWithEmailAndPassword(
-        email: _email, password: _password);
+        email: email, password: password);
 
     _firestore.collection('users').doc(result.user!.uid).set({
       'uid': result.user!.uid,
       'email': result.user!.email,
-      'location': GeoPoint(40.4440279, -79.9700647)
+      'location': const GeoPoint(40.4440279, -79.9700647)
     }, SetOptions(merge: true));
```
Comment (jeffzheng13) [inline lib/services/auth_service.dart:54]:
```suggestion
    }, SetOptions(merge: true));
    
    return userFromFirebaseUser(result.user!.uid);
```

Comment (jeffzheng13) [reply lib/services/auth_service.dart:54]: needs to return or else if we go to profile we get null for user.

File: lib/data/app_user.dart (hunk 1 above)
Comment (tbeidlershenk) [reply lib/data/app_user.dart:36]: Good idea

File: lib/services/auth_service.dart (hunk 3 above)
Comment (tbeidlershenk) [reply lib/services/auth_service.dart:54]: is that why that was happening??

**Review by jeffzheng13 — APPROVED** [review]
Good to go. Merging..

**Pull request merged**